src/MongoDB.Driver.Core/Core/Compression/Snappy/Snappy32NativeMethods.cs

```diff
@@ -107,14 +108,14 @@ private class SnappyLocator : RelativeLibraryLocatorBase
         {
             public override bool IsX32ModeSupported => true;
 
-            public override string GetLibraryRelativePath(SupportedPlatform currentPlatform)
+            public override string GetLibraryRelativePath(OperatingSystemPlatform currentPlatform)
```

Comment:
This change is not really required, but I thing having 2 classes to describe operating system options is too many :)

Reply:
A similar idea for all compression changes

Reply:
What do you mean by 2 classes?

Aren't you simply renaming the enum from SupportedPlatform to OperatingSystemPlatform?

Reply:
yes, but it's an internal class which is also required for MongoDb.Drivers (not only for core). So instead of creating two different classes, I created one shared.

Reply:
Maybe it makes sense to rename `OperatingSystemPlatform` to `SupportedPlatform`. It will make the diff smaller